corehq/apps/ota/models.py

```diff
@@ -198,3 +198,8 @@ class DeviceLogRequest(models.Model):
     username = models.CharField(max_length=255)
     device_id = models.CharField(max_length=255)
     created_on = models.DateTimeField(auto_now_add=True)
+
+    def save(self, *args, **kwargs):
+        from corehq.apps.ota.views import get_device_log_requests
```

Comment:
It seems less than ideal to have to do this import here. What do you think of moving `get_device_log_requests` to another module to remove the dependency involving `...ota.models` and `...ota.views`?

Reply:
I could stick it in `ota.utils`, though it'd still be a cyclical import (the getter needs the model, the model needs the getter).  I think the only other options would be moving cache invalidation to a manual call or putting the getter in this models file, neither of which I'm crazy about.

Actually, if I make `get_device_log_requests` a classmethod, I think `memoized` will still cache for the lifetime of the process, so that'd be a viable option too.  Thoughts?

Reply:
I agree, `memoized` `classmethod` seems like a good option.

Reply:
`get_device_log_requests` could also just be moved into `ota.models`. Doesn't look like it depends on anything in `ota.views`.

Reply:
ended up hiding this as an implementation detail and making a classmethod wrapper.

Reply:
👍 I like it